src/snapshot_middleware/mod.rs

```diff
@@ -388,3 +392,58 @@ pub fn default_sync_rules() -> &'static [SyncRule] {
         ]
     })
 }
+
+fn filter_default_property(
+    snapshot: &SyncbackSnapshot,
+    new_inst: &Instance,
+    name: &str,
+    value: &Variant,
+    attributes: &mut BTreeMap<String, UnresolvedValue>,
+    properties: &mut BTreeMap<String, UnresolvedValue>,
+) {
+    let db = rbx_reflection_database::get();
+    let class_descriptor = db.classes.get(new_inst.class.as_str());
+
+    match value {
+        Variant::Attributes(attrs) => {
+            for (attr_name, attr_value) in attrs.iter() {
+                // We (probably) don't want to preserve internal attributes,
+                // only user defined ones.
+                if attr_name.starts_with("RBX") {
+                    continue;
+                }
+                attributes.insert(
+                    attr_name.clone(),
+                    UnresolvedValue::from_variant_unambiguous(attr_value.clone()),
+                );
+            }
+        }
+        Variant::SharedString(_) => {
+            log::warn!(
+                "Rojo cannot serialize the property {}.{name} in JSON files.\n\
+                 If this is not acceptable, resave the Instance at '{}' manually as an RBXM or RBXMX.",
+                new_inst.class, snapshot.get_new_inst_path(new_inst.referent())
+            )
```

Comment:
Should this warning also provide the path of the file where the instance is defined?

Reply:
Do you mind if we tackle that in a future PR? Making that change effects several tests and I'd rather not clutter this PR with them

Reply:
Yeah, we can handle that in a different PR. A bit out of scope for this one either way.

Reply:
If we allow rbx-dom's rbx_reflector to write defaults for SharedString properties (which is now possible after rojo-rbx/rbx-dom#414, then we won't have to alter any test data

Reply:
I don't think there's a compelling reason to *not* have rbx-dom write SharedString defaults now. Seems like a good change.